### DISCUSSION TITLE
Is it possible to receive a string that contain accented characters ?

### DISCUSSION THREAD

**Discussion by MGoyheneche** [orchetect/swift-osc#37] [General]
Hi, 

I am currently struggling receiving message with accented characters.
Is there something I am missing ?

By the way, thanks for this cool lib !
Mathias :)

**Comment by orchetect** [comment]
As per the Open Sound Control specification, strings must be ASCII encoded.

That allows for certain accented characters but it's limited.

**Reply by MGoyheneche** [reply]
Hi, 
Thank you for your answer !
Do you how I can check if the string I receive is ASCII encoded ?
Thank you in advance.

**Reply by orchetect** [reply]
OSCKit forces strings that are sent and received to be encoded as ASCII. If unsupported characters are present they will either be lossily converted or the string may fail to send.

Unlike UTF8 or other modern encodings, ASCII cannot encode any Unicode character. It can represent only 256 specific characters. You can look up the ASCII table online to see what they are. Or you can experiment with `ASCIIString` and `ASCIICharacter` - two structs provided by OSCKit to do conversions.

**Reply by MGoyheneche** [reply]
Hi !
Thanks for your answer. I'm will try to understand why I don't receive OSC messages that have accented characters.
Have a nice day !
